lib/api.js

```diff
@@ -64,6 +64,41 @@ class API {
     })
   }
 
+  /**
+   * Updates an existing installation of the framework
+   * @param {Object} options
+   * @param {string} [options.version=null] Specific version of the framework to install
+   * @param {string} [options.repository] URL to github repo
+   * @param {string} [options.cwd=process.cwd()] Directory to install into
+   * @return {Promise}
+   */
+  async updateFramework ({
+    version = null,
+    repository = ADAPT_FRAMEWORK,
+    cwd = process.cwd(),
+    logger
+  } = {}) {
+    // cache state of plugins, as these will be wiped
+    const plugins = (await new Project({ cwd }).getInstallTargets())
+      .map(p => `${p.name}@${p.isLocalSource ? p.sourcePath : p.requestedVersion}`)
+      .map(p => p.isLocalSource ? p.sourcePath : `${p.name}@${p.requestedVersion}`)
```

Comment:
Did you mean to double map here? The second map is not performing on the strings returned by the initial map but it expecting plugin instances still. It looks like two ways of doing the same thing?

Reply:
Oh damn, no I think that first one should have been removed